tests/decorators/test_security.py

```diff
@@ -34,3 +36,137 @@ def func():
 
     with pytest.raises(OAuthProblem):
         wrapped_func()
+
+
+def _setup_request_mock(remote_addr, x_forwarded_for=None):
+    def get_header(header):
+        if header == "Authorization":
+            return "Bearer 12345678"
+        elif header == "X-Forwarded-For":
+            return x_forwarded_for
+    headers = MagicMock(side_effect=get_header)
+    request = MagicMock()
+    type(request).url = PropertyMock(return_value="http://localhost:9090/v1/resource/123")
+    type(request).remote_addr = PropertyMock(return_value=remote_addr)
+    return request, headers
+
+
+def _setup_token_request_mock(valid=True, status_code=200, text="OK"):
+    token_request = MagicMock()
+    type(token_request.return_value).ok = PropertyMock(return_value=valid)
+    type(token_request.return_value).status_code = PropertyMock(return_value=status_code)
+    type(token_request.return_value).text = PropertyMock(return_value=text)
+    return token_request
+
+
+def test_correct_client_ip_logged(monkeypatch):
+    def func():
+        pass
+
+    app = MagicMock()
+    monkeypatch.setattr('flask.current_app', app)
+
+    # 1) remote_addr, no x-forwarded-for, invalid token (401)
+    with LogCapture() as l:
+        wrapped_func = verify_oauth('https://example.org/tokeninfo', set(['admin']), [], func)
+        request, headers = _setup_request_mock(remote_addr="123.123.123.123")
```

Comment:
Better than doing all this mocking is to use `test_request_context`. It is safer because we relay on how Flask actually works and not how the mocks are set up.

```
api_client.application.test_request_context('/').push()
```
Reference:
- http://flask.pocoo.org/docs/0.12/reqcontext/
- http://flask.pocoo.org/docs/0.12/api/#flask.Flask.test_request_context